submissions/alstep07/task-js-post-oop/index.js

```diff
@@ -0,0 +1,116 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Inhabitant {
+	constructor(species, name, gender, saying, friends = []) {
+		this.species = species;
+		this.name = name;
+		this.gender = gender;
+		this.saying = saying;
+		this.friends = friends;
+	}
+
+	toString() {
+		let greating = [
+			`${this.saying}`,
+			`I am a ${this.species}`,
+			`My gender is ${this.gender}`,
+			`My name is ${this.name}`,
+		];
+
+		if (this.friends.length > 1) {
+			greating.push(`My friends are ${this.friends.join(", ")}`);
+		} else if (this.friends.length === 1) {
+			greating.push(`My friend is ${this.friends}`);
+		}
+		return greating.join(". ");
+	}
+}
+
+class Animal extends Inhabitant {
+	constructor(species, name, gender, saying, friends) {
+		super(species, name, gender, saying, friends);
+		this.legs = 4;
```

Comment:
No good reasons for a default value here. Four-legged animals have no preference compared with two-legged.

So `legs` is a shared property. Thus it has receive its initial value via `constructor` and child classes have no good reasons to assign it directly.

This way code will become more scalable and maintainable.

Reply:
@OleksiyRudenko  Thank you, I will fix it now.